### ISSUE TITLE
Flag to wait till deployment actually succeeded

### ISSUE THREAD

**Issue by yellowmegaman** [hashicorp/nomad#4606]
Hello there! Question here.

It would be really great if i could run `nomad run` and it won't release terminal till deployment is not only scheduled, but also completed successfully.

Use case:
- all deployments are being run from CI
- template nomad job against some environment via consul-template
- or template nomad job via terraform (template_file)
- plan/run template to deploy stuff
- terminal gets released if constrains/capacity are OK
- run script to check whenever deployment is successful by getting desiredCount and comparing to healthyCount in a loop

This approach is sub-optimal since it relies on some scripting which can be faulty and may have some corner cases.

Can we add flag to `nomad run` so it won't throw `0` exit code till deployment is finished?
Same would be awesome for nomad provider for terraform.

Thanks for attention in advance!

**Comment by jippi** [comment]
a work-around for now is to use [jrasell/levant](https://github.com/jrasell/levant)

**Comment by yellowmegaman** [comment]
@jippi thanks! seen it before, never had a chance to use it yet.

In my opinion it's not something that should be external to nomad, since it is already aware of the state of deployed services.

Anyway, gonna try levant for now, leaving issue open in case there would be some update whenever feature may land on roadmap for some future release.

**Comment by jippi** [comment]
Yes, I'm pretty sure Nomad will eventually support this - should be pretty trivial to add - just wanted to unblock you until Nomad got native support  :)

**Comment by yellowmegaman** [comment]
@jippi in that you've succeeded =) Thank again

**Comment by yellowmegaman** [comment]
Well, levant seems to be completely broken, deploying nomad job with broken on purpose healthcheck results in success, so hope this issue will see some attention someday.

**Comment by jrasell** [comment]
@yellowmegaman feel free to raise an issue against Levant and I can take a look into it. We use it extensively and successfully so I would be curious what the issue is.

**Comment by dadgar** [comment]
@yellowmegaman I do think it is a good feature and would want to add it. @jrasell PRs welcome 😄

**Comment by yellowmegaman** [comment]
@dadgar that is super-awesome =)
@jippi https://github.com/jrasell/levant/issues/227

**Comment by cgbaker** [comment]
@yellowmegaman : FYI, i'm adding this feature to the Nomad TF provider: 
https://github.com/terraform-providers/terraform-provider-nomad/issues/56

**Comment by pznamensky** [comment]
Looks like levant is abandoned by owner. Thus the flag in the nomad would be super useful for us too.

**Comment by pznamensky** [comment]
Is there any news here? :)
Unfortunately it blocks upgrade to nomad 1.0 for us.
We used to watch for deployments using levant, but it doesn't have hcl2 support (https://github.com/hashicorp/levant/issues/400)

**Comment by tcurdt** [comment]
No hcl2 support is a blocker to use levant.
And I also agree this should be ideally be built into nomad.

A `nomad run` should have an interactive option. Either interactive by default with a background/daemon option (see `docker-compose -d`) or non-interactive by default and allowing to activate interactive (see `docker -i`).

Not seeing allocation and deployments details as well as process stdout/err  on a `nomad run` is a terrible UX.
Worse - there is not even a simple way to get to this information like that from the command line.
Why is there no `nomad alloc list` for example?

**Comment by thatsk** [comment]
we want to see this feature in nomad cli. So that we can make ci-cd via nomad cli

**Comment by tgross** [comment]
This issue is a duplicate of https://github.com/hashicorp/nomad/issues/6818, which is on the roadmap. I'm going to close this issue in lieu of https://github.com/hashicorp/nomad/issues/6818 only because it has more of the implementation details on how we want to proceed on this. Keep an eye there for future progress (also: our Product folks do look at emoji reactions to issues, so feel free to add them there!)
